src/cljdoc/server/pedestal.clj

```diff
@@ -98,11 +100,12 @@
 
                 :cljdoc/index
                 (assoc ctx ::releases (storage/all-distinct-docs store))))})
-   (pu/html (fn [ctx]
+   (pu/body (fn data [ctx] (-> ctx ::releases index-pages/versions-tree))
+            (fn html [ctx versions-tree]
```

Comment:
I think the way this is done usually is that you just set `:response :body` to whatever data would be returned as JSON or EDN and then you provide an extra function that turns that into an html string. 

There's some more reference material here: http://pedestal.io/guides/hello-world-content-types#_returning_to_content_types

Reply:
Maybe if implemented properly we could refactor a bunch of our helpers that way.